packages/formatter-excel/src/excel.ts

```diff
@@ -0,0 +1,192 @@
+/**
+ * @fileoverview The most basic formatter, it just stringifyes whatever object
+ * is passed to it.
+ */
+
+/*
+ * ------------------------------------------------------------------------------
+ * Requirements
+ * ------------------------------------------------------------------------------
+ */
+import * as Excel from 'exceljs';
+import * as forEach from 'lodash.foreach';
+import * as groupBy from 'lodash.groupby';
+import * as sortBy from 'lodash.sortby';
+
+import { debug as d } from 'sonarwhal/dist/src/lib/utils/debug';
+import { IFormatter, Problem } from 'sonarwhal/dist/src/lib/types';
+import * as logger from 'sonarwhal/dist/src/lib/utils/logging';
+
+const _ = {
+    forEach,
+    groupBy,
+    sortBy
+};
+const debug = d(__filename);
+const startRow = 5;
+
+/*
+ * ------------------------------------------------------------------------------
+ * Formatter
+ * ------------------------------------------------------------------------------
+ */
+
+export default class ExcelFormatter implements IFormatter {
+    public async format(messages: Array<Problem>, target: string) {
+        if (messages.length === 0) {
+            return;
+        }
+
+        const resources: _.Dictionary<Array<Problem>> = _.groupBy(messages, 'resource');
+        const workbook = new Excel.Workbook();
+        const sortedResources = _.sortBy(Object.keys(resources));
+
+        // Styles for the cells
+        const bold = { font: { bold: true } };
+        const mediumBorder = { style: 'medium' };
+        const border = {
+            border: {
+                bottom: mediumBorder,
+                left: mediumBorder,
+                right: mediumBorder,
+                top: mediumBorder
+            }
+        };
+        const rightAlign = { align: { horizontal: 'right' } };
+        const tableHeader = Object.assign({
+            fill: {
+                fgColor: { argb: 'FF000000' },
+                pattern: 'solid',
+                type: 'pattern'
+            },
+            font: {
+                bold: true,
+                color: { argb: 'FFFFFFFF' }
+            }
+        }, border);
+
+        /**
+         * Replaces `://`, `/`, `.` for `-` so the names are compatible with the sheets
+         */
+        const getName = (name: string) => {
+            return name
```

Comment:
why don't use just one regex? something like `/:\/\/|[\/\.-]/g`

Reply:
I think it's clearer what's going on this way

Reply:
That's way you have a comment few lines above :P

Reply:
What the comment says (which is mainly for intellisense) and what the code actually does can be 2 completely different things 🙄
Also it will still need 2 steps because there's usually a trailing `/` that needs to be replaced with an empty string instead of a `-`.